include/effects/matrix/PatternAnimatedGIF.h

```diff
@@ -0,0 +1,299 @@
+#include <string.h>
+#include <ledstripeffect.h>
+#include <ledmatrixgfx.h>
+#include <ArduinoJson.h>
+#include "systemcontainer.h"
+#include <map>
+#include "effects.h"
+#include "types.h"
+#include <GifDecoder.h>
+
+// The GIF files are embedded within the flash image, and we need to tell the linker where they are
+
+extern const uint8_t atomic_start[]          asm("_binary_assets_gif_atomic_gif_start");
+extern const uint8_t atomic_end[]            asm("_binary_assets_gif_atomic_gif_end");
+extern const uint8_t colorsphere_start[]     asm("_binary_assets_gif_colorsphere_gif_start");
+extern const uint8_t colorsphere_end[]       asm("_binary_assets_gif_colorsphere_gif_end");
+
+// AnimatedGIFs
+//
+// Our set of embedded GIFs.  Currently assumed to be 32x32 in size, default FPS.
+
+enum
+{
+    INVALID     = 0,
+    Atomic      = 1,
+    ColorSphere = 2
+} GIFIdentifier;
+
+static std::map<int, EmbeddedFile, std::less<int>, psram_allocator<std::pair<int, EmbeddedFile>>> AnimatedGIFs =
+{
+    { Atomic, EmbeddedFile(atomic_start, atomic_end) },
+    { ColorSphere, EmbeddedFile(colorsphere_start, colorsphere_end) },
+};
+
+// For now, all gifs are assumed to be 32x32 in size.  We could make this more flexible, but it's not clear
+// that we need to.  If we do, we'll need to move the width and height into the AnimatedGIFs map so that we
+// can look them up by index.
+
+constexpr auto GIF_WIDTH = 32;
+constexpr auto GIF_HEIGHT = 32;
+
+static_assert(GIF_WIDTH <= MATRIX_WIDTH, "GIF_WIDTH must be <= MATRIX_WIDTH");
+static_assert(GIF_HEIGHT <= MATRIX_HEIGHT, "GIF_HEIGHT must be <= MATRIX_HEIGHT");
+
+// The decoder needs us to track some state, but there's only one instance of the decoder, and 
+// we can't pass it a pointer to our state because the callback doesn't allow you to pass any
+// context, and you can't use a lambda that captures the this pointer because that can't be
+// converted to a callback function pointer.  So we have to use a global. 
+
+struct 
+{
+    long            _seek      = 0;
+    const uint8_t * _pgif      = nullptr;
+    long            _len       = 0;
+    int             _offsetX   = 0;
+    int             _offsetY   = 0;
+    int             _skipCount = 0;
+    CRGB            _bkColor   = CRGB::Black;
+    CRGB            _skipColor = CRGB::Magenta;
+} 
+g_gifDecoderState;
+
+// We dynamically allocate the GIF decoder because it's pretty big and we don't want to waste the base
+// ram on it.  This way it, and the GIFs it decodes, can live in PSRAM.
+
+std::unique_ptr<GifDecoder<MATRIX_WIDTH, MATRIX_HEIGHT, 12>> g_ptrGIFDecoder = make_unique_psram<GifDecoder<MATRIX_WIDTH, MATRIX_HEIGHT, 12>>();
+
+// PatternAnimatedGIF
+//
+// Draws a cycling animated GIF on the LED matrix.  Use GifDecoder to do the heavy lifting behind the scenes.
+
+class PatternAnimatedGIF : public LEDStripEffect
+{
+private:
+    
+    int       _gifIndex = -1;
+    CRGB       _bkColor = BLACK16;
+    CRGB     _skipColor = MAGENTA16;

```

Comment:
I think the "de facto standard" within NDS has become to left-align the variable names, then add spaces where needed to align the equal signs, just like you do in `OpenGif()` further down:
```suggestion
    int  _gifIndex  = -1;
    CRGB _bkColor   = BLACK16;
    CRGB _skipColor = MAGENTA16;
```